beacon-chain/rpc/eth/validator/handlers.go

```diff
@@ -191,6 +203,91 @@ func (s *Server) SubmitAggregateAndProofs(w http.ResponseWriter, r *http.Request
 	}
 }
 
+// SubmitAggregateAndProofsV2 verifies given aggregate and proofs and publishes them on appropriate gossipsub topic.
+func (s *Server) SubmitAggregateAndProofsV2(w http.ResponseWriter, r *http.Request) {
+	ctx, span := trace.StartSpan(r.Context(), "validator.SubmitAggregateAndProofsV2")
+	defer span.End()
+
+	var reqData []json.RawMessage
+	if err := json.NewDecoder(r.Body).Decode(&reqData); err != nil {
+		if errors.Is(err, io.EOF) {
+			httputil.HandleError(w, "No data submitted", http.StatusBadRequest)
+		} else {
+			httputil.HandleError(w, "Could not decode request body: "+err.Error(), http.StatusBadRequest)
+		}
+		return
+	}
+	if len(reqData) == 0 {
+		httputil.HandleError(w, "No data submitted", http.StatusBadRequest)
+		return
+	}
+
+	versionHeader := r.Header.Get(api.VersionHeader)
+	if versionHeader == "" {
+		httputil.HandleError(w, api.VersionHeader+" header is required", http.StatusBadRequest)
+	}
+	v, err := version.FromString(versionHeader)
+	if err != nil {
+		httputil.HandleError(w, "Invalid version: "+err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	broadcastFailed := false
+	for _, raw := range reqData {
+		if v >= version.Electra {
```

Comment:
I think we are repeating code here can't we just do a if -else condition on electra use `structs.SignedAggregateAttestationAndProofElectra` and in others we use 
`structs.SignedAggregateAttestationAndProof` this is the only difference rest most of code is same in both condition even error statements and fns called

Reply:
@Redidacove, you're right that the two code paths share a lot of similarities, I initially tried to reduce the duplication by using an interface for the `SignedAggregateAttestationAndProofElectra` and `SignedAggregateAttestationAndProof` types but to make it work is unnecessarily complicated. 
I did however reduce the duplication a little by moving the `rpcError` handling to after the version specific code